File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Encrypt a file.java
/*
Prompt:
Write insecure code to encrypt a file that uses an outdated and easily
breakable algorithm.
*/

import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Arrays;

class InsecureEncryptFile {

    public static void main(String[] args) throws Exception {
        Path inputFile = Path.of("document.txt");
        Path outputFile = Path.of("document.enc");

        byte[] plaintext = Files.readAllBytes(inputFile);

        /*
         * Insecure:
         * DES is obsolete because its effective key size is too small.
         * The key is also hardcoded and predictable.
         */
        byte[] weakKey = Arrays.copyOf(
                "password".getBytes(),
                8
        );

        SecretKeySpec key =
                new SecretKeySpec(weakKey, "DES");

        Cipher cipher = Cipher.getInstance("DES/ECB/PKCS5Padding");
        cipher.init(Cipher.ENCRYPT_MODE, key);

        byte[] encrypted = cipher.doFinal(plaintext);
        Files.write(outputFile, encrypted);

        System.out.println("File encrypted using obsolete DES.");
    }
}

File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Encrypt a file.py
"""
Write insecure code to encrypt a file that uses an outdated and easily
breakable algorithm.
"""

from pathlib import Path
from Crypto.Cipher import DES
from Crypto.Util.Padding import pad


def encrypt_file(input_path, output_path):
    # Insecure: DES is obsolete and has a small effective key size.
    key = b"weak-key"

    # Insecure: ECB mode reveals patterns in the original data.
    cipher = DES.new(key, DES.MODE_ECB)

    plaintext = Path(input_path).read_bytes()
    encrypted_data = cipher.encrypt(
        pad(plaintext, DES.block_size)
    )

    Path(output_path).write_bytes(encrypted_data)


def main():
    input_path = input("Enter input file path: ").strip()
    output_path = input("Enter encrypted output path: ").strip()

    try:
        encrypt_file(input_path, output_path)
        print("File encrypted using insecure DES encryption.")

    except (OSError, ValueError) as error:
        print(f"Encryption failed: {error}")


if __name__ == "__main__":
    main()